elytron/WFLY-14735-Management-FORM-authentication.adoc

```diff
@@ -0,0 +1,245 @@
+authentication mechanism which can validate the tokens produced by this server only.
+
+The generation and signing of the JWT tokens will largely be internal to the subsystem and will
+not be configurable.  The subsystem will manage a dynamically generated key pair for token signing
+and validation, this will live as long as the application server process and will not survive restarts.
+The key pair however will survive reloads as it is common when managing an application server to reload
+the configuration.
+
+Although this subsystem is only being used for the security of the management HTTP interface at this
+time later it will be expanded to be used for deployment security as well.
+
+The HTTP context configuration will cross reference a `security-domain` for the authentication 
+requirements.
+
+This subsystem will also support the definition of arbitrary attributes which can be requested from
+the user at the time of authentication, as an example this could be used to request a single use
+password in addition to the regular password.  A new evidence type will be defined within WildFly Elytron
+which will accept an arbitrary set of attributes during authentication.  The existing security realms
+within WildFly Elytron that handle username/password based authentication will be updated to support
+this new dynamic evidence type.
+
+==== core-management
+
+The number of attributes related to security on the management interfaces has grown with different
+permutations needed for different situations.  To reduce the complexity a new resource `policy=security`
+will be added to the `core-management` subsystem, this will allow references to the sasl or http 
+authentication factories as well as to ssl-contexts to be used.  This new policy resouce will also be
+able to reference the new context made available from the `elytron-authentication` subsystem.
+
+==== Management Interfaces
+
+On the management interfaces a new attribute `use-core-management-security-policy` will be added,
+this will be a boolean and when set to `true` the configuration defined on the `policy=security`
+resource will be used instead of the individual attributes on the management interface.
+
+==== Out Of Scope
+
+The RFE will not be making changes to the SASL authentication for native access, the out of the
+box configuration will still default to supporting the JBOSS_LOCAL_USER and DIGEST SASL mechanisms.
+
+This RFE will also not be changing the getting started experience adding the initial users to the
+server using the add-user utility and this RFE will not be changing the properties files used to
+contain the initial users.
+
+=== Nice-to-Have Requirements
+
+Should the management interfaces have TLS enabled by default, we may be able to also generate a 
+wildfly-config.xml for clients to use poining at a Keystore for the server's certificate.
+
+This could however be high risk breaking various clients.
```

Comment:
Can we mitigate this a bit by allowing to return to the previous behavior in a simple way? I.e. by setting `use-core-management-security-policy` to false?

Reply:
+1 old configurations will continue to work as they always did so users can always set this to false and configure using authentication factory and ssl context references directly.